packages/synapse-react-client/src/components/SynapseTable/SynapseTableRenderers.tsx

```diff
@@ -221,12 +219,12 @@ export function TableDataColumnHeader(
   const facets = data?.facets ?? []
   const { getColumnDisplayName, getHelpText } = useQueryVisualizationContext()
 
-  if (!selectColumn) {
-    return <>{column.id}</>
-  }
-
-  const displayColumnName = getColumnDisplayName(selectColumn.name)
-  const columnHelpText = getHelpText(selectColumn.name)
+  const displayColumnName = selectColumn
+    ? getColumnDisplayName(selectColumn.name)
+    : column.id
+  const columnHelpText = selectColumn
+    ? getHelpText(selectColumn.name)
+    : undefined
```

Comment:
Small changes so that the code continues to always render the `ColumnHeader` component below (instead of rendering just the `column.id` value)